internal/v1alpha2/controllers/bundledeployment/bundledeployment.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bundledeployment
+
+import (
+	"context"
+	"sync"
+
+	"github.com/operator-framework/rukpak/api/v1alpha2"
+	"github.com/operator-framework/rukpak/internal/v1alpha2/store"
+	"github.com/spf13/afero"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	apimacherrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	crcontroller "sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	unpackpath = "/var/cache/bundles"
+)
+
+// BundleDeploymentReconciler reconciles a BundleDeployment object
+type bundleDeploymentReconciler struct {
+	client.Client
+
+	Scheme   *runtime.Scheme
+	Recorder record.EventRecorder
+
+	controller        crcontroller.Controller
+	dynamicWatchMutex sync.RWMutex
+	dynamicWatchGVKs  map[schema.GroupVersionKind]struct{}
+}
+
+//+kubebuilder:rbac:groups=core.rukpak.io,resources=bundledeployments,verbs=list;watch
+//+kubebuilder:rbac:groups=core.rukpak.io,resources=bundledeployments/status,verbs=update;patch
+//+kubebuilder:rbac:groups=core.rukpak.io,resources=bundledeployments/finalizers,verbs=update
+//+kubebuilder:rbac:groups=*,resources=*,verbs=*
```

Comment:
uh oh :(

Reply:
I know! But I have no idea on the intention behind this. Copied from https://github.com/operator-framework/rukpak/blob/bcae27f71c82b396dcf141b4fc557f06a089712d/internal/controllers/bundledeployment/bundledeployment.go#L160